### PR TITLE
Implement a check for whether we posted MassPay

### PR DESCRIPTION
**Pull request by whit537**
Hopefully will prevent recurrences of https://github.com/gratipay/inside.gratipay.com/issues/951#issuecomment-268832559.

### CODE REVIEW

**Comment by whit537** [comment]
Build is failing on #4236.

**Comment by whit537** [comment]
Base changed to #4236, builds restarted.

**Comment by mattbk** [comment]
#4236 is merged, should base be changed back to master?

**Comment by whit537** [comment]
Yup! Done and rebased. Previous was 20156d0ffd2dcb2fb466c0106e33f7816abe1a31.

**Comment by mattbk** [comment]
This looks okay to me as far as I can follow it. @nobodxbodon?

**Comment by whit537** [comment]
Rebased again.

File: www/dashboard/nmasspays.spt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+nmasspays = website.db.one("""
+
+    WITH payday AS (SELECT ts_start FROM paydays ORDER BY ts_start DESC LIMIT 1)
+
+    SELECT count(*)
+      FROM exchanges e
+      JOIN exchange_routes er
+        ON e.route = er.id
+     WHERE er.network = 'paypal'
+       AND e.amount < 0
```
Comment (nobodxbodon) [inline www/dashboard/nmasspays.spt]: what's this mean?

File: bin/masspay.py [hunk 2]
```diff
@@ -193,6 +193,11 @@ def post_back_to_gratipay():
     except KeyError:
         gratipay_base_url = 'https://gratipay.com'
 
+    nmasspays = int(requests.get(gratipay_base_url + '/dashboard/nmasspays').text())
+    if nmasspays < 10:
```
Comment (nobodxbodon) [inline bin/masspay.py]: so this assumes there should be more than 10 teams to pay to? any way to override this without changing code?

Comment (whit537) [inline bin/masspay.py]: Why would we want to override it?

File: www/dashboard/nmasspays.spt (hunk 1 above)
Comment (whit537) [inline www/dashboard/nmasspays.spt]: The semantics of the `exchanges.amount` field are somewhat goofy (especially as it interacts with the `fee` field; see #2442), but fundamentally a negative amount indicates a payout, whereas a positive amount indicates a payin.

File: bin/masspay.py (hunk 2 above)
Comment (nobodxbodon) [inline bin/masspay.py]: In case that there was no more than 10 teams to pay out to. Or is that impossible?

Comment (whit537) [inline bin/masspay.py]: Hmm ... theoretically, I suppose. We haven't been below 10 in a long time that I can recall (famous last words). This the test I came up with to decide whether masspay had been run the previous week or not. I would say in the case where it _was_ run but there were less than 10 payouts, the person running payday could adapt to the situation by manually modifying the script before rerunning. That should be a rare occurrence.

Comment (nobodxbodon) [inline bin/masspay.py]: Sounds good. Would it be helpful to add some instructions in the print out below, like "if this is a false alarm, please modify xxx manually and rerun"?

Comment (whit537) [inline bin/masspay.py]: How about 2de1d6689e361351004da6a0c22740b8a726c7bc?

**Comment by whit537** [comment]
Rebased, was 2de1d6689e361351004da6a0c22740b8a726c7bc.

File: www/dashboard/nmasspays.spt (hunk 1 above)
Comment (nobodxbodon) [inline www/dashboard/nmasspays.spt]: Thanks for explaining. Shall we document this in schema.sql (not necessary in this PR)?

**Comment by whit537** [comment]
Rebased, was 45c791ffa2abe9b91abc5f67f6140a0465c62d40. Ready to merge!

File: www/dashboard/nmasspays.spt (hunk 1 above)
Comment (whit537) [inline www/dashboard/nmasspays.spt]: Maybe on http://gratipay.readthedocs.io/?

**Pull request merged**